### PR TITLE
Fixes usage of HBFUtils::nOrbitsPerTF in TOF (e.g. raw data generation with non-standard) 

### PR DESCRIPTION
**Pull request by shahor02**
Hi @noferini 
Since you were caching the HBFUtils::getNOrbitsPerTF in  ``Geo::ORBIT_IN_TF``, the eventual changes in the former were not propagating to the code which needed this parameter.
This PR changes the caching to direct use of HBFUtils. Also, it allows to pass the ``configKeyValues`` option to TOF workflow, needed e.g. to set non-standard values in HBFUtils. No I am at least able to generate raw data with e.g. 128 HB per TF.

Still, I see some problems in  the raw data generation, please see comments I am adding to this PR.

### CODE REVIEW

**Comment by shahor02** [comment]
@noferini One more point: you are using direct access to RDH data members, which are version dependent. If you want to be version independent, you can apply modifications similar to [these](https://github.com/AliceO2Group/AliceO2/pull/3408/files#diff-5f8ff6e3e440973c35d66ab5f54942fc)

**Comment by shahor02** [comment]
@noferini can you please check this?

File: Detectors/TOF/base/include/TOFBase/WindowFiller.h [hunk 1]
```diff
@@ -35,7 +36,13 @@ class WindowFiller
 
   Int_t getCurrentReadoutWindow() const { return mReadoutWindowCurrent; }
   void setCurrentReadoutWindow(Double_t value) { mReadoutWindowCurrent = value; }
-  void setEventTime(double value) { mEventTime = value - mTF * Geo::NS_IN_TF; }
+  void setEventTime(double value)
+  {
+    mEventTime = value;
+    if (mTF) {
```
[the remark this answers is not in the document]
Comment (noferini) [reply Detectors/TOF/base/include/TOFBase/WindowFiller.h:42]: Yes. I took it out from the digitizer but I agree we can remove it also from here

[the remark this answers is not in the document]
Comment (noferini) [reply Detectors/TOF/base/include/TOFBase/WindowFiller.h:42]: Yes. I took it out from digitizer in the past. I agree we can remove it also from here

File: Detectors/TOF/reconstruction/src/Encoder.cxx [hunk 2]
```diff
@@ -262,7 +262,7 @@ bool Encoder::encode(std::vector<std::vector<o2::tof::Digit>> digitWindow, int t
   mEventCounter = tofwindow; // tof window index
   mIR.orbit = mEventCounter / Geo::NWINDOW_IN_ORBIT;
 
-  if (!(mIR.orbit % Geo::ORBIT_IN_TF)) { // new TF
+  if (!(mIR.orbit % mHBFSampler.getNOrbitsPerTF())) { // new TF
```
[the remark this answers is not in the document]
Comment (noferini) [reply Detectors/TOF/reconstruction/src/Encoder.cxx:265]: HBT are filled also for empty data.
About the usage of HBFUtils::fillHBIRvector, I can use it. In the past I didn't know where to take the first IT and then I assumed 0/0 to continue the implentation of raw encoding

**Pull request merged**